File: sample-projects/java/src/main/java/de/sots/cellarsandcentaurs/domain/model/ArmorClass.java
package de.sots.cellarsandcentaurs.domain.model;

import de.sots.cellarsandcentaurs.application.CreatureUtil;

public class ArmorClass {
    private final String description;
    private final int base;
    private final int bonus;

    public ArmorClass(int base, int bonus) {
        this(base, bonus, CreatureUtil.STANDARD_ARMOR_CLASS_DESCRIPTION);
    }

    public ArmorClass(int base, int bonus, String description) {
        this.description = description;
        this.base = base;
        this.bonus = bonus;
    }

    public int getTotal() {
        return base + bonus;
    }

    public String getDescription() {
        return description;
    }
}

File: sample-projects/cpp/src/de/sots/cellarsandcentaurs/domain/model/ArmorClass.cpp
#include "ArmorClass.hpp"

#include "../../application/Application.hpp"

namespace de::sots::cellarsandcentaurs::domain::model {

ArmorClass::ArmorClass(int base, int bonus)
    : ArmorClass(base, bonus, application::CreatureUtil::STANDARD_ARMOR_CLASS_DESCRIPTION) {}

ArmorClass::ArmorClass(int base, int bonus, std::string description)
    : description_{std::move(description)}, base_{base}, bonus_{bonus}, total_{base + bonus} {}

void ArmorClass::setBase(int base) noexcept {
    base_ = base;
    total_ = base_ + bonus_;
}

void ArmorClass::setBonus(int bonus) noexcept {
    bonus_ = bonus;
    total_ = base_ + bonus_;
}

}
